mechanicalsoup/stateful_browser.py

```diff
@@ -288,12 +288,13 @@ def _find_link_internal(self, link, args, kwargs):
 
         # Check if "link" parameter should be treated as "url_regex"
         # but reject obtaining it from both places.
-        if link and 'url_regex' in kwargs:
-            raise ValueError('link parameter cannot be treated as '
-                             'url_regex because url_regex is already '
-                             'present in keyword arguments')
-        else:
-            kwargs['url_regex'] = link
+        if link:
+            if 'url_regex' in kwargs:
+                raise ValueError('link parameter cannot be treated as '
+                                 'url_regex because url_regex is already '
+                                 'present in keyword arguments')
+            else:
+                kwargs['url_regex'] = link
```

Comment:
I'd rather keep the `if ... and ...` as is and turn `else:` into `else if link: ...` for readability.